resources/qml/ui/TimelineEffects.qml

```diff
@@ -0,0 +1,111 @@
+// SPDX-FileCopyrightText: Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import QtQuick 2.15
+import QtQuick.Particles 2.15
+
+Item {
+    readonly property int maxLifespan: Math.max(confettiEmitter.lifeSpan, rainfallEmitter.lifeSpan)
```

Comment:
This seems to not interact well with confetti spammed quickly? Or maybe the reason is somewhere else, but sometimes I just get no confetti :D

Reply:
IIRC I have some logic somewhere that dedups effects so you don't get lots of confetti in your face if you get several confetti events at once (e.g. opening an unread room). Probably that is what's happening.

Reply:
Maybe it is just an issue when testing stuff like this then. But I sent them a few seconds apart and only one worked, which was weird. :D